syncer/syncer.go

```diff
@@ -217,9 +218,13 @@ func (s *Syncer) closeJobChans() {
 	if s.jobsClosed.Get() {
 		return
 	}
+
+	s.jobsChanLock.Lock()
```

Comment:
whether needing to put this before `if s.jobsClosed.Get()` and put `s.jobsChanLock.Unlock()` after `s.jobsClosed.Set(true)`? and then `sync2.AtomicBool` can be changed to `bool`?

now, our concurrency control code is naive, we should refine this when doing the refactor.

Reply:
oh yes, we should lock the whole `closeJobChans` function